vulcanai2/datasets/tabulardataset.py

```diff
@@ -73,35 +106,60 @@ def __getitem__(self, idx):
             xs = torch.tensor(xs, dtype=torch.float)
             return xs
 
-    # TODO: needs to be re-written to match new stitch datasets signature.
-    # def merge_data(self, data, join_column, index_list=None, na_values=None):
-    #     """
-    #     Given additional data merge into the current data.
-    #     :param data: Either a path to a csv file, a dictionary names: paths to csv files, a list of dataframes, or a dataframe.
-    #     :param join_column: Either a single value if it is consistent, or a list of length data.
-    #     :param index_list: list of feature columns to index on when stitching(default None)
-    #     :return: None
-    #     """
-    #     if isinstance(data, list):
-    #         if not join_column:
-    #             raise RuntimeError("You need to provide a join_column if a list of csvs or dataframes are provided")
-    #         if isinstance(data[0], str):
-    #             dfs = [pd.read_csv(f, na_values=na_values) for f in data]
-    #         if isinstance(index_list, list):
-    #             assert len(data) == len(index_list)
-    #         self.df = utils.stitch_datasets(dfs, join_column)
-    #     elif isinstance(data, pd.DataFrame):
-    #         self.df = utils.stitch_datasets([self.df, data], join_column, index_list)
-    #     else:
-    #         new_df = pd.read_csv(data, na_values=na_values)
-    #         self.df = utils.stitch_datasets([self.df, new_df], join_column, index_list)
-    #     logger.info(f"Datasets successfully merged")
+    def merge_data(self, merge_on_columns=None,
+                   index_list=None, na_values=None, **kwargs):
+        """
+        Merges additional data into a TabularDataset isntance
+
+        Parameters
+        ----------
+        merge_on_columns: list of strings
+            Key(s) that specifies which columns to use to uniquely stitch
+            dataset (default None)
+        index_list: list of strings
+            List of columns to make the index of the dataframe
+        na_values: The values to convert to NaN when reading from csv
+        kwargs: keyword parameter, value is either path or dataframe
+            Where key: dataset name and value is either a path to a file
+            or a dataframe.
+        """
+
+        dataset_dict = kwargs
+
+        if "original" in dataset_dict:
```

Comment:
How do you know that 'original` will be in the dict?

Reply:
I'm checking to make sure it won't be because I want to name the original dataset "original". Not great, but Josephs scheme requires a name. Other suggestions appreciated!

Reply:
perhaps we can simply name them with a zero-indexed int in ascending order if no name is provided?

Reply:
Well:
a.) you can't not provide a name when merging as we're using keyword arguments
b.) you can't start a pandas column with an integer
c.) we don't have any way of knowing what was used before, which makes me a bit wary of using a, b, c as those are frequently used in variable names (which also negates using "original" if multiple merges are performed)

Reply:
@joemehltretter what are your thoughts on this. Is naming it "original" the best course of action?

Reply:
yeah I think that sounds good. If @nirtiac agrees, I think it would merit to prototype this and see what it looks like in practice

Reply:
I like it